pkg/web/oauthclient/oauthclient.go

```diff
@@ -0,0 +1,109 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauthclient
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	echo "github.com/labstack/echo/v4"
+	"go.thethings.network/lorawan-stack/pkg/component"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+// Config is the configuration for the OAuth client.
+type Config struct {
+	AuthorizeURL string `name:"authorize-url" description:"The OAuth Authorize URL"`
+	TokenURL     string `name:"token-url" description:"The OAuth Token Exchange URL"`
+
+	ClientID     string `name:"client-id" description:"The OAuth client ID"`
+	ClientSecret string `name:"client-secret" description:"The OAuth client secret" json:"-"`
+
+	StateCookieName string
```

Comment:
Add `name:"-"` to `StateCookieName` and `AuthCookieName`. Those don't need to be in flag config.